cli/replicasets/setup.go

```diff
@@ -0,0 +1,287 @@
+package replicasets
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"reflect"
+	"sort"
+
+	"github.com/apex/log"
+	"github.com/tarantool/cartridge-cli/cli/common"
+	"github.com/tarantool/cartridge-cli/cli/context"
+	"github.com/tarantool/cartridge-cli/cli/project"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	vshardRouterRole = "vshard-router"
+)
+
+type ReplicasetConf struct {
+	Alias         string   `yaml:"alias,omitempty"`
+	InstanceNames []string `yaml:"instances"`
+	Roles         []string `yaml:"roles"`
+
+	Weight      *float64 `yaml:"weight,omitempty"`
+	AllRW       *bool    `yaml:"all_rw,omitempty"`
+	VshardGroup *string  `yaml:"vshard_group,omitempty"`
+}
+
+type ReplicasetsConf map[string]*ReplicasetConf
+
+type ReplicasetsList []*ReplicasetConf
+
+func Setup(ctx *context.Ctx, args []string) error {
+	var err error
+
+	if err := FillCtx(ctx); err != nil {
+		return err
+	}
+
+	if ctx.Replicasets.File == "" {
+		ctx.Replicasets.File = defaultReplicasetsFile
+	}
+	if ctx.Replicasets.File, err = filepath.Abs(ctx.Replicasets.File); err != nil {
```

Comment:
Does this function return an error when file doesn't exist?

If answer "yes", it's non-informative to see `Failed to get replicasets configuration file absolute path` instead of `file "123.yml" doesn't exist`

Reply:
No, this function just computes a path (using simple string-concatenate rules). `doesn't exists` error is returned later on trying to open this file.

Reply:
Is there test for this case?

Reply:
Added a test.